libglusterfs/src/inode.c

```diff
@@ -598,18 +594,24 @@ inode_t *
 inode_unref(inode_t *inode)
 {
     inode_table_t *table = NULL;
+    xlator_t *xl = THIS;
```

Comment:
You don't need to fetch THIS -  only in line 607.

Reply:
I did it this way to entirely remove fetching `THIS` inside the critical section. `THIS` calls `__glusterfs_this_location` so it would be evaluated inside the lock itself if we call `__inode_unref(..., THIS)`. Is it okay to evaluate inside the lock?

Reply:
Indeed, it'd be great to fetch it outside the lock, but you can do it just before entering the lock, not right away unconditionally - as you might be exiting the function (say, it's a root node) without needing it at all. Some of my ongoing work is to reduce the unconditional fetch of 'THIS' everywhere in our code.

Reply:
This still need to be adjusted, if possible.